### PR TITLE
Access Graph AWS sync docs

### PR DESCRIPTION
**Pull request by tigrato**
This PR adds the required documentation to setup AWS Sync with Teleport Access Graph.

Closes https://github.com/gravitational/access-graph/issues/550

### CODE REVIEW

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 1]
```diff
@@ -0,0 +1,219 @@
+---
+title: Discover AWS access patterns with Teleport Access Graph
+description: Describes how to import and visualize AWS accounts access patterns using Teleport Access Graph.
+---
+
+The Teleport Access Graph offers invaluable insights into access patterns within
```
Comment (zmb3) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:6]:
```suggestion
The Teleport Access Graph offers insights into access patterns within
```

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+Utilizing the Access Graph to analyze IAM permissions within an AWS
+account necessitates the setup of the Teleport Access Graph (TAG)
+service, a Discovery Service, and integration with your AWS account.
+
+Teleport Access Graph is a feature of the [Teleport
+Policy](https://goteleport.com/platform/policy/) product that is only available
+to Teleport Enterprise customers.
+
+After logging into the Teleport UI, go to the Management tab. If enabled,
```
Comment (zmb3) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:24]:
```suggestion
After logging in to the Teleport UI, go to the Management tab. If enabled, 
```

https://helpcenter.veeam.com/docs/styleguide/tw/log_in_to.html

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+The importing process involves two primary steps:
+
+### Polling Cloud APIs
+
+The Teleport Discovery Service plays a pivotal role in this process
+by continuously scanning the configured AWS accounts. 
+At intervals of 15 minutes, it retrieves the following resources from your
+AWS account:
+
+- AWS Users
```
Comment (zmb3) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:44]: Do we need the `AWS` prefix on each of these?

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+ensuring that the Access Graph remains updated with the latest
+information from your AWS environment.
+
+### Importing resources
+
+Teleport Access Graph delves into the IAM policies, identities,
+and resources retrieved from your AWS account, crafting a 
+graphical representation thereof.
+
+Throughout this procedure, Access Graph seamlessly aligns
```
Comment (zmb3) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:64]: This sentence sounds like a sales pitch. I don't think it provides any value to someone looking to understand how things work or how to set it up.

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+their corresponding AWS IAM identities. This integration
+provides an enriched perspective, facilitating a deeper
+comprehension of users' AWS access patterns, especially
+when utilizing one of the nodes as a jump host.
+
+
+## Prerequisites
+
+- A running Teleport Enterprise cluster v14.3.9/v15.2.0 or later.
+- An updated `license.pem` with Teleport Policy enabled.
```
Comment (zmb3) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:75]: Should we mention this only applies to self-hosted clusters?

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+- Docker version v(=docker.version=) or later.
+- A running Teleport Access Graph node v1.7.0 or later.
+- The node running the Access Graph service must be reachable
+from Teleport Auth Service and Discovery Service. Check 
+[Access Graph page](../access-graph.mdx) for details on
+how to setup Teleport Access Graph.
+
+## Step 1/2. Configure Discovery Service (On-Prem only)
```
Comment (zmb3) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:83]:
```suggestion
## Step 1/2. Configure Discovery Service (Self-hosted only)
```

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+
+<Notice type="warning">
+
+If you have a Teleport Cloud cluster, please disregard
+this step, as Teleport Cloud already operates a properly configured 
+Discovery Service within your cluster.
+</Notice>
+
+To activate the Teleport Discovery Service, it's necessary to
```
Comment (zmb3) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:92]:
```suggestion
To activate the Teleport Discovery Service,
```

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+- Access Graph service is reachable from the machine where Discovery Service runs.
+
+## Step 2/2. Set up Access Graph AWS Sync
+
+To initiate the setup wizard for configuring AWS Sync, access the Teleport UI,
+navigate to the Management tab, and choose the Access Graph option within the
+Permission Management section.
+
+If both Teleport and Access Graph support AWS sync, you'll notice a new button
+adjacent to the Access Graph navigation bar labeled `Setup AWS sync with
```
Comment (zmb3) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:120]:
```suggestion
adjacent to the Access Graph navigation bar labeled `Set up AWS sync with
```
(Please fix the UI too. Setup is a noun, set up is a verb)

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+Access Graph`.
+
+The setup process is intuitive, guiding you through the steps. You'll be
```
Comment (zmb3) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:123]:
```suggestion
You'll be
```

Remove extra fluff

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 10]
```diff
@@ -0,0 +1,219 @@
...
+</Details>
+
+
+Once the IAM Policy has been successfully linked to the IAM role
+utilized by Teleport, you'll be prompted to specify the regions from
+which Teleport should import resources. This selection solely pertains
+to regional resources and does not impact global resources such as S3
+Buckets, IAM Policies, or IAM Users.
+
+If you're operating an on-premises cluster, you'll additionally need to
```
Comment (zmb3) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:216]:
```suggestion
If you're operating a self-hosted cluster, you'll additionally need to
```

Most people running Teleport in their own AWS environments would not consider that to be "on-premises" as it's an Amazon datacenter, not their own.

File: docs/pages/access-controls/access-graph/aws-sync.mdx (hunk 3 above)
Comment (tigrato) [reply docs/pages/access-controls/access-graph/aws-sync.mdx]: Fixed in 8926e7fd8b

File: docs/pages/access-controls/access-graph/aws-sync.mdx (hunk 4 above)
Comment (tigrato) [reply docs/pages/access-controls/access-graph/aws-sync.mdx]: Removed it.

File: docs/pages/access-controls/access-graph/aws-sync.mdx (hunk 5 above)
Comment (tigrato) [reply docs/pages/access-controls/access-graph/aws-sync.mdx]: fixed in 8926e7fd8b

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+- A running Teleport Enterprise cluster v14.3.9/v15.2.0 or later.
+- For self-hosted clusters, an updated `license.pem` with Teleport Policy enabled.
+- Docker version v(=docker.version=) or later.
+- A running Teleport Access Graph node v1.7.0 or later.
+- The node running the Access Graph service must be reachable
+from Teleport Auth Service and Discovery Service. Check 
+[Access Graph page](../access-graph.mdx) for details on
+how to setup Teleport Access Graph.
+
+## Step 1/2. Configure Discovery Service (Self-hosted only)
```
Comment (jakule) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:76]: This page is listed only for `"enterprise"`, so it won't show for the cloud anyway. We should either add the cloud section to it or remove non-self-hosted only sections.

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 12]
```diff
@@ -0,0 +1,212 @@
...
+and resources retrieved from your AWS account, crafting a 
+graphical representation thereof.
+
+
+## Prerequisites
+
+- A running Teleport Enterprise cluster v14.3.9/v15.2.0 or later.
+- For self-hosted clusters, an updated `license.pem` with Teleport Policy enabled.
+- Docker version v(=docker.version=) or later.
+- A running Teleport Access Graph node v1.7.0 or later.
```
Comment (jakule) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:70]:
```suggestion
- A running Teleport Access Graph node v1.17.0 or later.
```

Comment (jakule) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:70]: I'd also mention that you need to configure Teleport and Access Graph and link the docs with instructions

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 13]
```diff
@@ -0,0 +1,212 @@
...
+As part of the setup process, you'll be prompted to execute a bash script
+within your AWS Cloud Shell. This script's purpose is to generate an IAM
+Policy within your AWS account and link it to the AWS Role designated for
+your Teleport AWS Integration. The policy is designed with a set of read-only
+actions, enabling Teleport to access and retrieve information from resources
+within your AWS Account.
+
+The IAM policy includes the following directives:
+
+
```
Comment (jakule) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:134]: nit:
```suggestion
```

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 14]
```diff
@@ -0,0 +1,212 @@
...
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        // EC2 IAM
```
Comment (jakule) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:143]: JSON doesn't support comments. I'd either remove them to keep valid JSON or not use JSON here.

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 15]
```diff
@@ -0,0 +1,212 @@
...
+To initiate the setup wizard for configuring AWS Sync, access the Teleport UI,
+navigate to the Management tab, and choose the Access Graph option within the
+Permission Management section.
+
+If both Teleport and Access Graph support AWS sync, you'll notice a new button
+adjacent to the Access Graph navigation bar labeled `Set up AWS sync with
+Access Graph`.
+
+You'll be
+prompted to create a new Teleport AWS integration if you haven't configured
```
Comment (jakule) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:117]: nit:
```suggestion
You'll be prompted to create a new Teleport AWS integration if you haven't configured
```

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 16]
```diff
@@ -0,0 +1,212 @@
...
+one already. Alternatively, you can opt for a previously established integration.
+
+Upon selecting or creating the integration, you'll be instructed to execute a
+bash script within your AWS Cloud Shell to configure the necessary permissions.
+
+<Details title="List of IAM Policies required for AWS Sync" closed>
+
+As part of the setup process, you'll be prompted to execute a bash script
+within your AWS Cloud Shell. This script's purpose is to generate an IAM
```
Comment (jakule) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:126]: Those two sentences are basically the same.

**Review by jakule — APPROVED** [review]

File: docs/pages/access-controls/access-graph/aws-sync.mdx (hunk 11 above)
Comment (tigrato) [reply docs/pages/access-controls/access-graph/aws-sync.mdx:76]: Forgot that. Fixed

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 17]
```diff
@@ -0,0 +1,199 @@
...
+Teleport Access Graph is a feature of the [Teleport
+Policy](https://goteleport.com/platform/policy/) product that is only available
+to Teleport Enterprise customers.
+
+After logging in to the Teleport UI, go to the Management tab. If enabled, 
+Access Graph options can be found under the Permission Management section.
+
+## How AWS Sync Process works
+
+Teleport Access Graph seamlessly synchronizes various AWS resources,
```
Comment (ptgott) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:29]:
```suggestion
Teleport Access Graph synchronizes various AWS resources,
```

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 18]
```diff
@@ -0,0 +1,199 @@
...
+EKS clusters, and RDS databases. These resources are then visualized
+using the graph representation detailed in the
+[Access Graph page](../access-graph.mdx).
+
+The importing process involves two primary steps:
+
+### Polling Cloud APIs
+
+The Teleport Discovery Service plays a pivotal role in this process
+by continuously scanning the configured AWS accounts.
```
Comment (ptgott) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:40]:
```suggestion
The Teleport Discovery Service continuously scans the configured AWS accounts. 
```

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 19]
```diff
@@ -0,0 +1,199 @@
...
+Check [Access Graph page](../access-graph.mdx) for details on
+how to setup Teleport Access Graph.
+- The node running the Access Graph service must be reachable
+from Teleport Auth Service and Discovery Service. 
+
+## Step 1/2. Configure Discovery Service (Self-hosted only)
+
+<Notice type="warning">
+
+If you have a Teleport Cloud cluster, please disregard
```
Comment (ptgott) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:79]:
```suggestion
If you have a Teleport Cloud cluster, you can disregard
```
To be consistent with the rest of the docs

File: docs/pages/access-controls/access-graph/aws-sync.mdx [hunk 20]
```diff
@@ -0,0 +1,199 @@
...
+</Notice>
+
+To activate the Teleport Discovery Service,
+add the provided snippet to your Auth Service configuration. 
+This service monitors dynamic `discovery_config` resources
+that are set up with the `discovery_group` matching 
+<Var name="access-graph-disc" />.
+
+
+```code
```
Comment (ptgott) [inline docs/pages/access-controls/access-graph/aws-sync.mdx:91]:
```suggestion
```yaml
```

**Review by ptgott — APPROVED** [review]

**Pull request merged**